pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -24,25 +27,30 @@ import (
 const failedSetCPUManagerLabelFmt = "failed to set a cpu manager label on host %s"
 
 type HeartBeat struct {
-	clientset                 k8scli.CoreV1Interface
+	clientSet                 k8scli.NodeInterface
+	client                    kubecli.ShadowNodeInterface
 	deviceManagerController   device_manager.DeviceControllerInterface
 	clusterConfig             *virtconfig.ClusterConfig
 	host                      string
 	cpuManagerPaths           []string
 	devicePluginPollIntervall time.Duration
 	devicePluginWaitTimeout   time.Duration
+	nodeBackoff               bool
 }
 
-func NewHeartBeat(clientset k8scli.CoreV1Interface, deviceManager device_manager.DeviceControllerInterface, clusterConfig *virtconfig.ClusterConfig, host string) *HeartBeat {
+func NewHeartBeat(shadowClient kubecli.ShadowNodeInterface, nodeClient k8scli.NodeInterface, deviceManager device_manager.DeviceControllerInterface, clusterConfig *virtconfig.ClusterConfig, host string) *HeartBeat {
 	return &HeartBeat{
-		clientset:               clientset,
+		client:                  shadowClient,
+		clientSet:               nodeClient,
 		deviceManagerController: deviceManager,
 		clusterConfig:           clusterConfig,
 		host:                    host,
 		// This is a temporary workaround until k8s bug #66525 is resolved
 		cpuManagerPaths:           []string{virtutil.CPUManagerPath, virtutil.CPUManagerOS3Path},
 		devicePluginPollIntervall: 1 * time.Second,
 		devicePluginWaitTimeout:   10 * time.Second,
+
+		nodeBackoff: true,
```

Comment:
Add a comment on what this flag means or change it to a more intuitive name, such as updateNodeHeartBeat

Reply:
Agree